### PR TITLE
Disabled radio group should not be in tab order

### PR DESCRIPTION
**Pull request by hcarmona**
This change fixes an issue where a paper-radio-group will focus one of its disabled children if all children are disabled. See: http://crbug.com/642644

### CODE REVIEW

**Comment by danbeam** [comment]
/cc @notwaldorf @bicknellr @tjsavage 

this is for Chrome's Material Design Settings page
https://bugs.chromium.org/p/chromium/issues/detail?id=642644

and if we're trying to match native controls, at least in Chrome's case disable radios aren't focusable nor tabbable:
https://jsfiddle.net/d6vgaynt/

note: just settings `tabIndex = -1` leaves a node _focusable_ (as in: clicking shows a focus ring, `.focus()` still works)

File: paper-radio-group.html [hunk 1]
```diff
@@ -120,6 +119,12 @@
       allowEmptySelection: {
         type: Boolean,
         value: false
+      },
+
+      disabled: {
+        type: Boolean,
+        reflectToAttribute: true,
```
Comment (notwaldorf) [inline paper-radio-group.html]: Just double checking like a nagging person: are you styling this based on `disabled` anywhere? (like `paper-radio-group[disabled[ {...}`. If not, this doesn't need to reflect

**Comment by notwaldorf** [comment]
LGTM, small nagging question

**Comment by danbeam** [comment]
@notwaldorf asked me offthread if we could do this in IronControlState instead.

@hcarmona how much more work would this be?  it'd definitely apply to more places / probably affect more things

File: paper-radio-group.html (hunk 1 above)
Comment (danbeam) [inline paper-radio-group.html]: there's code in higher-up behaviors that check for `.hasAttribute('disabled')`, which is why I asked @hcarmona to add `reflectToAttribute`

**Comment by bicknellr** [comment]
If you disable the group, but not the contained buttons, are they still in the tab order?

**Comment by bicknellr** [comment]
Tried editing the demo, looks like IronMenuBehavior has a 'focus' handler that handles this.

**Comment by notwaldorf** [comment]
@danbeam not instead, alongside. I don't think this element has `IronControlState`

**Comment by notwaldorf** [comment]
@bicknellr I think this `disabled` attribute doesn't do anything without styling (it doesn't do anything magic from the platform)

**Comment by danbeam** [comment]
> If you disable the group but not the contained buttons, are they still in the tab order?

i'll let @hcarmona answer fully, but I believe we're also disabling the buttons ourselves:
https://cs.chromium.org/chromium/src/chrome/browser/resources/settings/people_page/sync_page.html?l=238,242

**Comment by danbeam** [comment]
> @danbeam not instead, alongside. I don't think this element has `IronControlState`

it's in `IronMenuBehavior`

**Comment by danbeam** [comment]
@bicknellr so maybe we can just add

```js
properties: {
  disabled: {
    type: Boolean,
    reflectToAttribute: true,
  },
},
```

and `IronMenuBehavior` will Just Work

**Comment by bicknellr** [comment]
Yeah, this fix looks like it would be better for `IronMenuBehavior`. After messing around with the demo some more I found that if you add `disabled` to the second paper-radio-group in the demo, you can still tab to the radio button.

**Comment by notwaldorf** [comment]
Russell's on this PR so i'ma run away and pass him the hot potato YAY 🍾

**Comment by bicknellr** [comment]
There's a function `_resetTabindices` there which you could modify pretty cleanly to get this behavior.

**Comment by hcarmona** [comment]
Changes do make more sense in the ``` IronMenuBehavior ``` 
The only change that I would make here would be to remove the tabindex from the hostAttributes. (After updating ``` IronMenuBehavior ```)

>  you add a test to make sure that if the radio group is disabled you can't select a child button? Or focus to it, etc

While writing a test, I found all the weird edge cases about selecting children.

> If you disable the group but not the contained buttons, are they still in the tab order?

There's a few cases we need to consider. Some don't seem valid IMO.

Easy cases:
Group disabled, all buttons disabled. Nothing is focus-able or tab-able. *This is what we want*
Group enabled, all buttons enabled. All buttons are focus-able and tab handling is normal (up/down)
Group enabled, some buttons disabled, only valid buttons selected (or no selection). Normal tab and focus handling.

Invalid case:
Group disabled, any button is enabled. Is there a use case for this, and what should happen?

Odd cases:
Group enabled, all buttons disabled. Group is focus-able and tab-able, but values can't be changed?
Group enabled, some buttons disabled, disabled button selected. Can the value be changed?

**Comment by danbeam** [comment]
what's the status on this PR?  is it ready to review?  is it blocked on the `IronMenuBehavior`

**Comment by hcarmona** [comment]
This is blocked on the IronMenuBehavior PR. ( https://github.com/PolymerElements/iron-menu-behavior/pull/88 ) Once that is landed, then we only need to remove the tabindex from here.

**Comment by ktiedt** [comment]
Semi related, `prg.disabled = true` should also play a role in disabling its radio controls no? Happy to open a separate issue, but figured I'd at least ask on this PR before doing so.

**Comment by hcarmona** [comment]
I had forgotten about this PR. Can it still be merged since PolymerElements/iron-menu-behavior#88 has been merged (it was the only blocking issue before)?

@ktiedt This PR works with the dependent PR to make sure that the radio buttons arent focusable when they're disabled. It doesn't affect how they're disabled.

**Comment by ktiedt** [comment]
@hcarmona glad I could help ;)

As for disabling, I understand this PR, was hoping maybe one of the Polymer folks would chime in on if thats expected or intentional -- seems the radio groups have had several questionable deficiencies come up lately.

**Comment by bicknellr** [comment]
We recently decided that we're going to run tests against internal stuff before merging on GitHub. I'll cc you both on the CL I'm using to do that.

**Comment by bicknellr** [comment]
https://go/polymerelements_paper-radio-group_pull_77
All of those failing are the same as those in other, unrelated train runs happening at the same time (example http://go/polymerelements_iron-iconset-svg_pull_81), so I'm going to call this ok.
